--- a/gui/postinstall.cmake
+++ b/gui/postinstall.cmake
@@ -1,4 +1,4 @@
-file(READ ${CMAKE_INSTALL_PREFIX}/share/applications/khopper.desktop desktop)
+file(READ $ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/share/applications/khopper.desktop desktop)
 string(REPLACE ";" "\;" desktop "${desktop}")
 string(REPLACE "__ICON__" "${CMAKE_INSTALL_PREFIX}/share/pixmap/khopper.png" desktop "${desktop}")
-file(WRITE ${CMAKE_INSTALL_PREFIX}/share/applications/khopper.desktop ${desktop})
+file(WRITE $ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/share/applications/khopper.desktop ${desktop})
